runtime-light/stdlib/file/file-system-functions.h

```diff
@@ -180,9 +181,408 @@ inline resource f$stream_socket_client(const string& address, std::optional<std:
 }
 
 inline Optional<string> f$file_get_contents(const string& stream) noexcept {
+  // TODO Why is empty `mode` passed here to f$fopen?
   if (auto sync_resource{from_mixed<class_instance<kphp::fs::sync_resource>>(f$fopen(stream, {}), {})}; !sync_resource.is_null()) {
     auto expected{sync_resource.get()->get_contents()};
     return expected ? Optional<string>{*std::move(expected)} : Optional<string>{false};
   }
   return false;
 }
+
+// TODO move write_safe() and read_safe() to utils.cpp or leave them here ??
+inline std::expected<size_t, int32_t> write_safe(kphp::fs::sync_resource* resource, std::span<const std::byte> src) {
+  /* TODO need tracing?
```

Comment:
Tracing won't be supported anytime soon. I don't think these comments are necessary.